src/codeswitch/data_generation/synthatic_data_generation.py

```diff
@@ -0,0 +1,160 @@
+import os
+import random
+import re
+import time
+from collections import Counter
+
+import pandas as pd
+from openai import OpenAI
+from tqdm import tqdm
+
+# pylint: disable=E1101
+
+# Initialize the OpenAI client
+api_key = os.getenv("open_ai_key")
+if not api_key:
+    raise ValueError("The open_ai_key environment variable is not set.")
+
+client = OpenAI(api_key=api_key)
+
+
+def classify_topic(utterance, retries=3, delay=2):
+    for _ in range(retries):
+        try:
+            stream = client.chat.completions.create(
+                model="gpt-3.5-turbo",
+                messages=[
+                    {
+                        "role": "system",
+                        "content": "You are a helpful assistant that classifies utterances into topics.",
+                    },
+                    {
+                        "role": "user",
+                        "content": f"Extract and return only the main topic from the utterance given: avoid any comments apart from the classified topic: '{utterance}'.",
+                    },
+                ],
+                stream=True,
```

Comment:
While streaming doesn't hurt, I'm not seeing how it is helpful in our case. It just complicates the logic.